pandas/core/dtypes/base.py

```diff
@@ -0,0 +1,89 @@
+"""Extend pandas with custom array types"""
+import abc
+
+
+class ExtensionDtype(metaclass=abc.ABCMeta):
+    """A custom data type for your array.
+    """
+    @property
+    def type(self):
+        """Typically a metaclass inheriting from 'type' with no methods."""
+        return type(self.name, (), {})
```

Comment:
I'm not really sure what `array.dtype.type` is used for. This passes the test suite, but may break things like

```
array1.dtype.type is array2.dtype.type
```

since the object IDs will be different (I think).

Reply:
In NumPy, `dtype.type` is the corresponding scalar type, e.g.,
```
>>> np.dtype(np.float64).type
numpy.float64
```

I don't know where "Typically a metaclass inheriting from 'type' with no methods." comes from.

Reply:
Thanks, that makes sense. Would you say that `object` is a good default here? I'll work through a test array that uses something meaningful like `numbers.Real` or `int`. We do use `values.dtype.type` in a few places, like figuring out which block type to use.

Reply:
I don't think there's a good default value here. Generally the right choice is to return the corresponding scalar type, e.g., `Interval` for `IntervalDtype`.